Quintana_Jerrod/Assignments/f+sql_projects/wall/server.py

```diff
@@ -0,0 +1,106 @@
+from flask import Flask, redirect, render_template, flash, session, request
+from mysqlconnection import MySQLConnector
+from flask.ext.bcrypt import Bcrypt
+import re
+app = Flask(__name__)
+mysql = MySQLConnector(app, 'login_registration')
+app.secret_key = 'ThisIsSecret'
+bcrypt = Bcrypt(app)
+email_reg = re.compile(r'^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9._-]+\.[a-zA-Z]+$')
+def validation():
+    if 'id' in session:
+        return True;
+    else:
+        return False;
+
+
+@app.route('/')
+def index():
+    return render_template('index.html')
+
+@app.route('/wall')
+def wall():
+    if (validation() == False):
+        flash('You need to be logged in to access this page')
+        return redirect('/')
+    all = mysql.query_db('SELECT messages.id id, messages.user_id as user_id, messages.message message, users.first_name first, users.last_name last, DATE_FORMAT(messages.created_at, "%b %d %Y %h:%i %p") date FROM messages JOIN users ON users.id = messages.user_id ORDER BY date DESC')
+    comment = mysql.query_db('SELECT comments.comment comment, users.first_name first, users.last_name last, comments.message_id cid, DATE_FORMAT(comments.created_at, "%b %d %Y %h:%i %p") date FROM comments JOIN users ON users.id = comments.user_id ORDER BY date DESC')
+    return render_template('wall.html', all = all, comment= comment)
+
+@app.route('/process', methods=['POST'])
+def register():
+    if request.form['action'] == 'register':
+        if len(request.form['first']) < 2 or request.form['first'].isalpha() == False:
+            flash('First name can only have letters and must be filled out')
+            return redirect('/')
+        if len(request.form['last']) < 2 or request.form['last'].isalpha() == False:
+            flash('Last name can only have letters and must be filled out')
+            return redirect('/')
+        if not email_reg.match(request.form['email']):
+            flash('Email must be in the proper example@example.com format')
+            return redirect('/')
+        if len(request.form['pwd']) < 8:
+            flash('Password must at least 8 characters long')
+            return redirect('/')
+        if request.form['pwd'] != request.form['confirm']:
+            flash("Password and Confirmation don't match")
+            return redirect('/')
+        check = mysql.query_db("SELECT email FROM users")
+        for i in check:
+            if request.form['email'] == i['email']:
+                flash('A user with that email already exists')
+                return redirect('/')
```

Comment:
check all validations before return redirect.  Currently the first time a validation fails it'll redirect before checking any others